apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2387,3 +2387,19 @@ func (u *UniterAPI) SetContainerSpec(args params.SetContainerSpecParams) (params
 	}
 	return results, nil
 }
+
+// GoalStates
+// TODO (agprado): Implement in the next PR
+func (u *UniterAPI) GoalStates(args params.Entities) (params.StringResults, error) {
+	results := params.StringResults{
```

Comment:
As discussed, there's a helper accessUnit() func defined on UniterAPI. it is created in the facade set up. You want to use that here.
See AssignedMachine() for an example.
You also need to use a for loop to iterate over the args. And parse the unit tag and get the unit etc. AssignedMachine() is a good example for that as well. Do that here and then you can return "Hello World <unitname>" in the result(s).

Reply:
Next change is to return a struct that can be parsed as yaml, or json. But now I have something that works, and returns the correct text.

    agprado @ agprado-Laptop: ~/Projects/Canonical/juju/bin: :
    $ ./juju run --unit mysql/0 'goal-state'
    Hello World